### PR TITLE
Build Roslyn.sln using dotnet

### PR DESCRIPTION
**Pull request by tmat**
Roslyn.sln can now be built using `dotnet build/pack`. VSIXes are not produced from such build.

Allows us to run test projects targeting .NET Core that are not in Compilers.sln.

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar @dotnet/roslyn-infrastructure PTAL

**Comment by tmat** [comment]
@jasonmalinowski

**Comment by jmarolf** [comment]
> VSIXes are not produced from such build.

I assume this is because we need .NET Core compatible versions of those targets?

**Comment by tmat** [comment]
@jmarolf Yes. https://devdiv.visualstudio.com/DevDiv/_git/VSExtensibility/pullrequest/227498

File: src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <PropertyGroup>
     <_NuGetRepackAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(NuGetPackageRoot)microsoft.dotnet.nugetrepack.tasks\$(MicrosoftDotnetNuGetRepackTasksVersion)\tools\net472\Microsoft.DotNet.NuGetRepack.Tasks.dll</_NuGetRepackAssembly>
-    <_NuGetRepackAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NuGetPackageRoot)microsoft.dotnet.nugetrepack.tasks\$(MicrosoftDotnetNuGetRepackTasksVersion)\tools\netcoreapp3.1\Microsoft.DotNet.NuGetRepack.Tasks.dll</_NuGetRepackAssembly>
+    <_NuGetRepackAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NuGetPackageRoot)microsoft.dotnet.nugetrepack.tasks\$(MicrosoftDotnetNuGetRepackTasksVersion)\tools\netcoreapp2.1\Microsoft.DotNet.NuGetRepack.Tasks.dll</_NuGetRepackAssembly>
```
Comment (jaredpar) [inline src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj:7]: Can you elaborate on why this change is needed?

**Comment by jaredpar** [comment]
Really excited about this. Would much prefer to just `dotnet build` all the time vs having to hunt down the proper `msbuild`. 

> VSIXes are not produced from such build.

What is the mechanism that makes this happen? I can see from the change why we don't produce the uber VSIX anymore but can't see why the individual VSIX are no longer building.

**Comment by tmat** [comment]
Arcade skips VSIX targets when building on .NET Core.

File: src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj (hunk 1 above)
Comment (tmat) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj:7]: microsoft.dotnet.nugetrepack.tasks package does not target `netcoreap3.1`. My guess is we have replaced `netcoreapp2.1` with `netcoreapp3.1` in all targets files.
